### ISSUE TITLE
MOV1 instruction cannot operate on the zero page

### ISSUE THREAD

**Issue by michaelcmartin** [kleinesfilmroellchen/spcasm#147]
Consider the following code, which spinlocks on the ENDX register until the sample in voice 0 finishes playing:

```
8F FC F2    LOOP: MOV $F2,#FC
AA F3 00          MOV1 C,$F3.0
90 F8             BCC LOOP
```

Actually feeding this code to spcasm does not produce this output, instead yielding this error:
> Invalid addressing mode `$F3.0` as second operand for `MOV1`
>
>  help: The instruction `MOV1`, with the first operand `C`, accepts the modes address, address.bit as second operands

Changing the address to a location in the range $0100-$1FFF allows assembly to proceed as normal.

I _suspect_ that what is happening here is that this argument is being interpreted as a `dp.bit` mode operand, and then the emitter is noticing that `MOV1` doesn't actually have that as a legal mode, and then gives up instead of promoting it to `address.bit`.

One other anomaly here is that it claims to accept `address` _alone_ as an operand, which it does and which it assembles it as if it were `address.1`. This behavior is neither in the ISA spec you cite nor the opcodes test cases.

**Comment by kleinesfilmroellchen** [comment]
Thanks for the report, very much appreciated! I have reproduced this issue on the online playground and will look at it with a local build later (the playground CI has been busted for various reasons for a while now, so it's not up-to-date). I want to quickly answer your suspicions beforehand, though.

> I _suspect_ that what is happening here is that this argument is being interpreted as a `dp.bit` mode operand, and then the emitter is noticing that `MOV1` doesn't actually have that as a legal mode, and then gives up instead of promoting it to `address.bit`.

This is most likely true; codegen can't change addressing modes anymore so it needs to be corrected beforehand.

> One other anomaly here is that it claims to accept `address` _alone_ as an operand,

You're probably misunderstanding the help message. The help message claims that *given the first operand `C`*, there's a certain list of second operands. So the help is indeed suggesting one correct addressing mode here; `address` alone should not be allowed on MOV1 since it cannot carry a bit index. (Some mnemonics can carry a bit index themselves, like SET, CLR, BBC, or BBS)

> which it does and which it assembles it as if it were `address.1`. This behavior is neither in the ISA spec you cite nor the opcodes test cases.

This is another, probably unrelated bug in the codegen tables.

**Comment by kleinesfilmroellchen** [comment]
Fix for the main issue is now in b0138183ff0e9d677ded7a48a94af56c2024b13f.

> One other anomaly here is that it claims to accept `address` _alone_ as an operand, which it does and which it assembles it as if it were `address.1`. This behavior is neither in the ISA spec you cite nor the opcodes test cases.

Looking at this again, I misunderstood your original comment. This is intended behavior and matches Asar behavior; part of the integration test suite includes the Asar opcode test itself. I should however document it.
